src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -18,15 +19,18 @@ internal class GenerateCommand : CommandBase
         public GenerateCommand(Application parent)
             : base(parent, "generate")
         {
-            Sources = Argument("sources", ".cshtml files to compile", multipleValues: true);
+            Sources = Option("-s", ".cshtml files to compile", CommandOptionType.MultipleValue);
+            Outputs = Option("-o", "Generated output file path", CommandOptionType.MultipleValue);
+            RelativePaths = Option("-r", "Relative path", CommandOptionType.MultipleValue);
             ProjectDirectory = Option("-p", "project root directory", CommandOptionType.SingleValue);
-            OutputDirectory = Option("-o", "output directory", CommandOptionType.SingleValue);
             TagHelperManifest = Option("-t", "tag helper manifest file", CommandOptionType.SingleValue);
         }
 
-        public CommandArgument Sources { get; }
+        public CommandOption Sources { get; }
 
-        public CommandOption OutputDirectory { get; }
+        public CommandOption Outputs { get; }
```

Comment:
I might be missing something but why do we need separate output file paths instead of just a directory?

Reply:
We want MSBuild to tell us where individual files go

Reply:
We want MSBuild to control all of the path computations. It's the ideal tool for that